Modules/xSharePoint/DSCResources/MSFT_xSPUserProfileSyncService/MSFT_xSPUserProfileSyncService.psm1

```diff
@@ -84,8 +89,13 @@ function Set-TargetResource
     Invoke-xSharePointCommand -Credential $FarmAccount -Arguments $PSBoundParameters -ScriptBlock {
         $params = $args[0]
         
-
-        $syncService = Get-SPServiceInstance -Server $env:COMPUTERNAME | Where-Object { $_.TypeName -eq "User Profile Synchronization Service" }
+        $currentServer = $env:COMPUTERNAME
+        $syncService = Get-SPServiceInstance -Server $currentServer | Where-Object { $_.TypeName -eq "User Profile Synchronization Service" }
+        if ($null -eq $syncService) { 
+            $domain = (Get-CimInstance -ClassName Win32_ComputerSystem).Domain
+            $currentServer = "$currentServer.$domain"
+            $syncService = Get-SPServiceInstance -Server $currentServer | Where-Object { $_.TypeName -eq "User Profile Synchronization Service" }
+        }
```

Comment:
What happens if after this change the result is still $null?? As far as I can see there is no additional check.